src/main/java/se/sundsvall/esigning/integration/esigning/util/EsigningMapper.java

```diff
@@ -0,0 +1,110 @@
+package se.sundsvall.esigning.integration.esigning.util;
+
+
+import static java.util.Collections.emptySet;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import se.sundsvall.esigning.api.model.Initiator;
+import se.sundsvall.esigning.api.model.Message;
+import se.sundsvall.esigning.api.model.Reminder;
+import se.sundsvall.esigning.api.model.Signatory;
+import se.sundsvall.esigning.api.model.SigningRequest;
+
+public final class EsigningMapper {
+
+	private EsigningMapper() {
+	}
+
+	public static generated.se.sundsvall.pw_e_signing.SigningRequest toSigningRequest(final SigningRequest signingRequest) {
+		if (signingRequest == null) {
+			return null;
+		}
+
+		final var newRequest = new generated.se.sundsvall.pw_e_signing.SigningRequest();
+
+		Optional.ofNullable(signingRequest.getLanguage()).ifPresent(newRequest::setLanguage);
+		Optional.ofNullable(signingRequest.getCallbackUrl()).ifPresent(newRequest::setCallbackUrl);
+		Optional.ofNullable(signingRequest.getExpires()).ifPresent(newRequest::setExpires);
+
+		Optional.ofNullable(signingRequest.getDocument()).ifPresent(document -> {
+			newRequest.setName(document.getDescriptiveName());
```

Comment:
Should this not follow the Optional.OffNullable pattern as this value is nullable?

Reply:
If the document.getDescriptiveName() returns null, the name is just set to null which it otherwise would be anyways. There is no risk of NPE here.

Reply:
The three top Optional chains has the same nullcheck and I cant see any difference between those cases and this?